enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,543 @@
+---
+title: operator-status
+authors:
+  - "@awgreene"
+reviewers:
+  - "@ecordell"
+approvers:
+  - "@kevinrizza"
+creation-date: 2020-05-19
+last-updated: 2020-06-10
+status: implementable
+see-also:
+  - "N/A"  
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# operator-status
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The goal of this enhancement is to allow operators managed by the [Operator Lifecycle Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) operator to communicate conditions and states that OLM would otherwise be unable to infer. The presence of certain conditions should influence OLM behavior to protect the cluster.
+
+## Motivation
+
+OLM currently infers the state of an operator by inspecting the Kubernetes resources that define the operator. OLM does not provide operators with the means to communicate complex states that cannot be inferred without operator specific knowledge.
+
+Operators managed by OLM would benefit from the ability to communicate their state directly to OLM through a supported channel. Operators could communicate complex conditions and influence how OLM manages the operator, dramatically improving user experience.
+
+A core requirement of this enhancement is to provide operators with the ability to communicate to OLM that the operator should not be upgraded while a critical process is being executed. Consider the following scenario based on OLM today:
+
+>An operator installed with automatic updates is executing a critical process that could place the cluster in a bad state if interrupted. While this critical process is running, a new version of the operator becomes available. Since the operator was installed with automatic updates, OLM upgrades the operator without allowing the critical process to finish and places the cluster in an unrecoverable state.
+
+OLM can provide a significantly better user experience by creating a channel for the operator to communicate state that OLM cannot infer otherwise.
+
+### Goals
+
+- Provide operators with the means to explicitly communicate a condition that OLM cannot infer on its own.
+- Allow operators to define specific conditions which will influence OLM behavior, such as the "Upgrade Readiness" condition defined above.
```

Comment:
nit: I don't see the "Upgrade Readiness" above this.

Reply:
I had imagined that "Upgrade Readiness" would be infered from [this passage](https://github.com/operator-framework/enhancements/pull/23/files#diff-e72af68c33ea1f2583161d8a322dbf62R39),  but will update the passage to read:

> A core requirement of this enhancement is to provide operators with the ability to communicate "Upgrade Readiness", allowing the operator to completes a critical process without being upgraded by OLM.

Reply:
Addressed in the latest release of this PR.